### DISCUSSION TITLE
[Projects] Add support for creating/updating Views and Fields programatically

### DISCUSSION THREAD

**Discussion by marosset** [community/community#41133] [Projects and Issues]
### Select Topic Area

Product Feedback

### Body

There should be a way to create and/or update Views or Fields in project boards programmatically.

The Kubernetes project has transitioned to using project boards for tracking various parts of the project (new enhancements, bug triage, etc) and many of these boards get newly created for each release. 
Trying to recreate all of the Views and Fields for multiple board for each release is tedious and error prone.

It would be nice if the GraphQL APIs had mutations to manipulate Views/Fields/etc or if there was a way to create a new project board from a template.

Example project boards:
- https://github.com/orgs/kubernetes/projects/98
- https://github.com/orgs/kubernetes/projects/80
- https://github.com/orgs/kubernetes/projects/117

**Comment by dtzar** [comment]
Tangent related issues on Kubernetes related to desired type of views for projects boards which aren't automatable, however it gives an idea of the types of boards which we would like to spread/implement programmatically or easily across numerous kubernetes project repos/teams without having to set all of them up.
https://github.com/kubernetes/community/issues/5535
https://github.com/kubernetes/test-infra/issues/9925
